### PR TITLE
Merge latest ExploitFixes.cpp updates

### PR DESCRIPTION
**Pull request by KittenPopo**
Commits from my fork added to this PR:
- https://github.com/R2Northstar/NorthstarLauncher/commit/11f04d371b4c572a30e6497c171f62a699f3be84
- https://github.com/R2Northstar/NorthstarLauncher/commit/a2b208a960e2b90ed534b380bda93486340ee7ef (not DLOG stuff)
- https://github.com/R2Northstar/NorthstarLauncher/commit/aff5ea1f4403155a20bdfae22c56b08cd4a11e79
- https://github.com/R2Northstar/NorthstarLauncher/commit/5b33e0b4181c4d46257228bad3b77f1326fd2e95

FEAR NOT: All new changes since my original PR to ExploitFixes (i.e. Script_GetEntByIndex hook) were added.

Should work just fine for everyone. Let me know if any changes are needed.

### CODE REVIEW

File: NorthstarDedicatedTest/ExploitFixes.cpp [hunk 1]
```diff
@@ -431,7 +469,4 @@ void ExploitFixes::LoadCallback(HMODULE baseAddress)
 
 	ns_exploitfixes_log =
 		new ConVar("ns_exploitfixes_log", "1", FCVAR_GAMEDLL, "Whether to log whenever ExploitFixes.cpp blocks/corrects something");
-
-	HookEnabler hook;
-	ENABLER_CREATEHOOK(hook, (char*)baseAddress + 0x2a8a50, &GetEntByIndexHook, reinterpret_cast<LPVOID*>(&GetEntByIndex));
```
Comment (GeckoEidechse) [inline NorthstarDedicatedTest/ExploitFixes.cpp:436]: This strips fixes by #191

**Review by GeckoEidechse — CHANGES_REQUESTED** [review]

File: NorthstarDedicatedTest/ExploitFixes.cpp (hunk 1 above)
Comment (cpdt) [reply NorthstarDedicatedTest/ExploitFixes.cpp:455]: It was just moved to further up, using KHOOK:

https://github.com/R2Northstar/NorthstarLauncher/blob/561f97c223c2a71355592116bfa638690aebf6d2/NorthstarDedicatedTest/ExploitFixes.cpp#L74-L81

File: NorthstarDedicatedTest/ExploitFixes.cpp [hunk 2]
```diff
@@ -5,11 +5,13 @@
 #include "NSMem.h"
 #include "cvar.h"
 
-typedef char(__fastcall* function_containing_emit_t)(uint64_t a1, uint64_t a2);
-function_containing_emit_t function_containing_emit;
-ConVar* sv_cheats;
+#define ENTITYLIST_SLOT_COUNT 4096
```
Comment (cpdt) [inline NorthstarDedicatedTest/ExploitFixes.cpp:8]: All of the (not broken) native code seems to use 16384 (`0x4000`) when checking bounds for the entity array. Where does 4096 come from?

File: NorthstarDedicatedTest/ExploitFixes.cpp (hunk 1 above)
Comment (GeckoEidechse) [reply NorthstarDedicatedTest/ExploitFixes.cpp:455]: Oh that's where it went. Thanks for linking <3

File: NorthstarDedicatedTest/ExploitFixes.cpp (hunk 2 above)
Comment (KittenPopo) [reply NorthstarDedicatedTest/ExploitFixes.cpp:8]: https://developer.valvesoftware.com/wiki/Entity_index

File: NorthstarDedicatedTest/ExploitFixes.cpp [hunk 3]
```diff
@@ -260,60 +270,29 @@ KHOOK(ReadUsercmd, ("server.dll", "4C 89 44 24 ? 53 55 56 57"), void, __fastcall
...
-	}
-
-	if (cmd->attackangles.IsInvalid())
-	{
-		BLOCKED_INFO("CMD has invalid attackangles");
-		goto INVALID_CMD;
-	}
+#define FORCE_3_VALID(name)                                                                                                                \
+	if (name.IsInvalid())                                                                                                                  \
+	name = {0, 0, 0}
```
Comment (GeckoEidechse) [inline NorthstarDedicatedTest/ExploitFixes.cpp:275]: Shouldn't `#defines` generally be at the top of the file for readability? ^^

Comment (KittenPopo) [reply NorthstarDedicatedTest/ExploitFixes.cpp:275]: It's for a specific function and is not used elsewhere. This is also a .cpp file, not a header, so it cannot (should not) be imported.
I can add #undef after the function or convert it to a template function if you'd like.

**Pull request closed without merging**